applications/matter_weather_station/sysbuild.cmake

```diff
@@ -0,0 +1,9 @@
+#
+# Copyright (c) 2023 Nordic Semiconductor ASA
+#
+# SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+#
+
+if("${OVERLAY_CONFIG}" STREQUAL "overlay-factory_data.conf")
+    set(PM_STATIC_YML_FILE ${APPLICATION_CONFIG_DIR}/pm_static_factory_data.yml PARENT_SCOPE)
+endif()
```

Comment:
This too, it would make more sense for this to be an application sysbuild Kconfig which, if set, will include the overlay and set the static PM file